### PR TITLE
tests: Added tests for suites

### PR DESCRIPTION
**Pull request by Harshil2107**
Change-Id: I69db8e82e9373d659d125d3bd48a69de12b32390

### CODE REVIEW

File: tests/gem5/suite_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Suite Tests
+
+These tests runs a workload from every suite in gem5 resources.
```
Comment (powerjg) [inline tests/gem5/suite_tests/README.md]: Does it run the workload to completion or just make sure the workload can be "loaded"?

File: tests/gem5/suite_tests/configs/suite_run_workload.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.isas import ISA
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+
+parser = argparse.ArgumentParser(description="A script to run suite tests.")
+
+parser.add_argument(
+    "-i", "--id", type=str, required=True, help="The suite id."
```
Comment (powerjg) [inline tests/gem5/suite_tests/configs/suite_run_workload.py]:
```suggestion
    "-i", "--suite-id", type=str, required=True, help="The suite id."
```

File: tests/gem5/suite_tests/configs/suite_run_workload.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    return processor
+
+
+if args.isa == "riscv":
+    requires(isa_required=ISA.RISCV)
+
+    from gem5.prebuilt.riscvmatched.riscvmatched_board import RISCVMatchedBoard
+
+    board = RISCVMatchedBoard()
+    is_fs = False
```
Comment (powerjg) [inline tests/gem5/suite_tests/configs/suite_run_workload.py]: Why are all RISC-V suites not full system?

File: tests/gem5/suite_tests/configs/suite_run_workload.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    from gem5.components.boards.x86_board import X86Board
+
+    processor = get_processor(ISA.X86)
+    board = X86Board(
+        clk_freq="1GHz",
+        processor=processor,
+        memory=memory,
+        cache_hierarchy=cache_hierarchy,
+    )
+    is_fs = True
```
Comment (powerjg) [inline tests/gem5/suite_tests/configs/suite_run_workload.py]: Why are all x86 suites full system?

File: tests/gem5/suite_tests/test_suite.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    isa: str,
+    version: Optional[str] = None,
+    to_tick: Optional[int] = None,
+):
+    name = f"suite-{id}_{isa}_suite_test"
+
+    verifiers = []
+
+    config_args = [
+        "--id",
```
Comment (powerjg) [inline tests/gem5/suite_tests/test_suite.py]:
```suggestion
        "--suite-id",
```

File: tests/gem5/suite_tests/test_suite.py [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            valid_isas=(constants.all_compiled_tag,),
+            valid_hosts=constants.supported_hosts,
+        )
+
+
+test_suite(
+    id="riscv-vertical-microbenchmarks",
+    isa="riscv",
+    to_tick=10000000000,
+)
```
Comment (powerjg) [inline tests/gem5/suite_tests/test_suite.py:100]: `--is-fs` should probably be an option as well.

File: tests/gem5/suite_tests/configs/suite_run_workload.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+)
+
+parser.add_argument(
+    "-s",
+    "--isa",
+    type=str,
+    required=True,
+    help="The ISA to use.",
+)
+
```
Comment (powerjg) [inline tests/gem5/suite_tests/configs/suite_run_workload.py:82]: You should probably make `--is-fs` an argument as well.

File: tests/gem5/suite_tests/configs/suite_run_workload.py (hunk 3 above)
Comment (Harshil2107) [reply tests/gem5/suite_tests/configs/suite_run_workload.py]: We only have 1 riscv suite and that is the microbenchmarks which is SE mode. I was think that we update this script when we have more suites.

File: tests/gem5/suite_tests/configs/suite_run_workload.py (hunk 4 above)
Comment (Harshil2107) [reply tests/gem5/suite_tests/configs/suite_run_workload.py]: The only 2 x86 suites are gapbs and npb which are fs.

Comment (powerjg) [reply tests/gem5/suite_tests/configs/suite_run_workload.py]: I think this should be an option and not hardcoded. I don't think it makes sense to hardcode whether the *suite* is fs or not in the run script. This should be information with the suite. In this case, it should be in the test description file.

**Comment by Harshil2107** [comment]
Note: The CI test is failing due to bug in riscvmatched board which is fixed in #677

File: tests/gem5/suite_tests/README.md (hunk 1 above)
Comment (Harshil2107) [reply tests/gem5/suite_tests/README.md]: This runs it to make sure the workload is loaded, it runs to 10 billion ticks.

Comment (powerjg) [reply tests/gem5/suite_tests/README.md]: You should update the description with that information 😄

**Comment by powerjg** [comment]
What's the status of this PR?

**Comment by Harshil2107** [comment]
I will look into fixing the failing test then it should be good to go, I think it is due to a resource issue

File: tests/gem5/suite_tests/configs/suite_run_workload.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    from gem5.components.boards.x86_board import X86Board
+
+    processor = get_processor(ISA.X86)
+    board = X86Board(
+        clk_freq="1GHz",
+        processor=processor,
+        memory=memory,
+        cache_hierarchy=cache_hierarchy,
+    )
+# Set the workload.
```
Comment (powerjg) [inline tests/gem5/suite_tests/configs/suite_run_workload.py:130]: Should have  a panic if it's on RISC-V or X86 (i.e., Arm)

File: tests/gem5/suite_tests/README.md [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# Suite Tests
+
+These tests runs one workload from every suite in gem5 resources tp 10 billion ticks.
```
Comment (powerjg) [inline tests/gem5/suite_tests/README.md]: Typo 

```suggestion
These tests runs one workload from every suite in gem5 resources to 10 billion ticks.
```

**Review by powerjg — APPROVED** [review]

**Review by BobbyRBruce — APPROVED** [review]

**Pull request merged**